src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -389,52 +397,161 @@ sub-<label>/[ses-<label>/]
-the FSL format for bvec specification is the fact that the coordinate system of
-the bvecs is with respect to the participant, defined by the axes of the
-corresponding `dwi.nii` file, and not the magnet's coordinate system.
-Thus, any rotations applied to `dwi.nii` also need to be applied to the
-corresponding bvec file.
-
-bvec example:
+distinguish different sets of parameters.
+
+The OPTIONAL [`dir-<label>`](../99-appendices/09-entities.md#dir)
+key/value pair corresponds to a custom label the user may use to
+distinguish different sets of phase-encoding directions.
+
+**Combining multi- and single-band acquisitions**.
+The single-band reference image MAY be stored with suffix `sbref` (for example,
+`dwi/sub-control01_sbref.nii[.gz]`) as long as the image has no corresponding
+[gradient information (`[*_]dwi.bval` and `[*_]dwi.bvec` sidecar files)](#required-gradient-orientation-information)
+to be stored.
+
+Otherwise, if some gradient information is associated to the single-band diffusion
+image and a multi-band diffusion image also exists, the `acq-<label>` key/value pair
+MUST be used to distinguish both images.
+In such a case, two files could have the following names:
+`sub-01_acq-singleband_dwi.nii.gz` and `sub-01_acq-multiband_dwi.nii.gz`.
+The user is free to choose any other label than `singleband` and
+`multiband`, as long as they are consistent across subjects and sessions.
+
+### REQUIRED gradient orientation information
+
+The REQUIRED gradient orientation information corresponding to a DWI acquisition
+MUST be stored using `[*_]dwi.bval` and `[*_]dwi.bvec` pairs of files.
+The `[*_]dwi.bval` and `[*_]dwi.bvec` files MAY be saved on any level of the directory structure
+and thus define those values for all sessions and/or subjects in one place (see
+[the inheritance principle](../02-common-principles.md#the-inheritance-principle)).
+
+As an exception to the [common principles](../02-common-principles.md#definitions)
+that parameters are constant across runs, the gradient table information (stored
+within the `[*_]dwi.bval` and `[*_]dwi.bvec` files) MAY change across DWI runs.
+
+**Gradient orientation file formats**.
+The `[*_]dwi.bval` and `[*_]dwi.bvec` files MUST follow the
+[FSL format](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FDT/UserGuide#DTIFIT):
+The `[*_]dwi.bvec` file contains 3 rows with *N* space-delimited floating-point numbers
+(corresponding to the *N* volumes in the corresponding NIfTI file.)
+The first row contains the *x* elements, the second row contains the *y* elements and
+the third row contains the *z* elements of a unit vector in the direction of the applied
+diffusion gradient, where the *i*-th elements in each row correspond together to
+the *i*-th volume, with `[0,0,0]` for *non-diffusion-weighted* (also called *b*=0 or *low-b*)
+volumes.
+Following the FSL format for the `[*_]dwi.bvec` specification, the coordinate system of
```

Comment:
The official FSL format for bvecs is more complicated than that and if the bvecs are truly aligned with the image axes many software packages will misinterpret the bvecs. For example an LPS+ image axis will result in flipped y bvecs in FSL and mrtrix, but not DSI Studio or Dipy.

It makes more sense to either say it's "similar to the FSL format, with bvecs aligned to the image axes" or to make it clear that bvecs are *exactly* the FSL format and will intrinsically flip bvecs for some non LAS+ image orientations. The latter seems definitely less BIDSy because it's additional complexity to meet what is arguably a bug in a specific software package rather than to simply describe the data. This problem goes away with the RASB file format in the other PR

Reply:
Actually! Yes, I wanted to flag this.   

I realise this is maybe out of scope of this PR, but note that Appendix 8 mentions 3 types of coordinate systems, "device coordinates" (scanner), "file format coordinates" (DICOM), annd "head" coordinates (for example, NIFTI).

Maybe I can offer a change that links up with the definitions in that appedix?  See next...

Reply:
Similarly, here I just wanted to:
- Add the `MultipartID` concept
- Refactor the specs of DWI to be more consistent with the functional MRI (a bit more developed than DWI for historical reasons) and making explicit some caveats very implicit in the current spec.

I think #352 provides a great opportunity to refine what is said about the FSL format, make a clear route for deprecation and resolve the problem it originally addressed.

Reply:
I have accepted the suggestion (adding a link to the coordinate systems is helpful and does not entail any big change to the gradients). That said, I still think we should make the deep revision of gradients in #352.

Reply:
sounds good. Let me know when you get to #352 and I am happy to help